sdk/monitor/monitor-opentelemetry-exporter/src/export/metric.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { diag } from "@opentelemetry/api";
+import {
+  AggregationTemporality,
+  InstrumentType,
+  PushMetricExporter,
+  ResourceMetrics,
+} from "@opentelemetry/sdk-metrics-base";
+import { ExportResult } from "@opentelemetry/core";
+import { AzureMonitorBaseExporter } from "./base";
+import { AzureExporterConfig } from "../config";
+import { TelemetryItem as Envelope } from "../generated";
+import { resourceMetricsToEnvelope } from "../utils/metricUtils";
+
+/**
+ * Azure Monitor OpenTelemetry Metric Exporter.
+ */
+export class AzureMonitorMetricExporter
+  extends AzureMonitorBaseExporter
+  implements PushMetricExporter
+{
+  /**
+   * Initializes a new instance of the AzureMonitorMetricExporter class.
+   * @param AzureExporterConfig - Exporter configuration.
+   */
+  constructor(options: AzureExporterConfig = {}) {
+    super(options);
+    diag.debug("AzureMonitorMetricExporter was successfully setup");
+  }
+
+  /**
+   * Export OpenTelemetry resource metrics.
+   * @param metrics - Resource metrics to export.
+   * @param resultCallback - Result callback.
+   */
+  async export(
+    metrics: ResourceMetrics,
+    resultCallback: (result: ExportResult) => void
+  ): Promise<void> {
+    diag.info(`Exporting ${metrics.scopeMetrics.length} metrics(s). Converting to envelopes...`);
+
+    let envelopes: Envelope[] = resourceMetricsToEnvelope(
+      metrics,
+      this._options.instrumentationKey
+    );
+    resultCallback(await this._exportEnvelopes(envelopes));
+  }
+
+  /**
+   * Shutdown AzureMonitorMetricExporter.
+   */
+  public async shutdown(): Promise<void> {
+    diag.info("Azure Monitor Trace Exporter shutting down");
+    return this._shutdown();
+  }
+
+  public selectAggregationTemporality(_instrumentType: InstrumentType) {
```

Comment:
can the input parameter be removed?

Reply:
OpenTelemetry interface is incorrect apparently, so I cannot remove it, aggregation temporality should be only part of MetricReaders and not MetricExporters, I was thinking to add some not implemented error but typescript complains about return type

Reply:
I meant removing the input parameter only and not the method, along the lines of `public selectAggregationTemporality() { return AggregationTemporality.CUMULATIVE; }`, see [playground example here](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgBrIN7JgLmQCgA88QBXAWwCNoBKZAXgD5kA3Ae2ABNkBfAWABQghABs4AZ3HIAmsmDkADiIjkI4KegyDkO7PjpaBu48m27+AnkA_)

Reply:
Got it, I didn't know you could do that, I though typescript would complain about something like that